src/txn/txn_rollback_to_stable.c

```diff
@@ -573,82 +577,19 @@ __rollback_abort_col_ondisk_kv(WT_SESSION_IMPL *session, WT_REF *ref, WT_COL *ci
-          __wt_timestamp_to_string(rollback_timestamp, ts_string[2]));
-        if (!F_ISSET(S2C(session), WT_CONN_IN_MEMORY))
-            /* Allocate tombstone and calls function add update to head of insert list. */
-            return (__rollback_col_ondisk_fixup_key(session, ref, rollback_timestamp, true, recno));
-        else {
-            /*
-             * In-memory database does not have a history store to provide a stable update, so
-             * remove the key.
-             */
-            WT_RET(__wt_upd_alloc_tombstone(session, &upd, NULL));
-            WT_STAT_CONN_DATA_INCR(session, txn_rts_keys_removed);
-        }
-    } else if (WT_TIME_WINDOW_HAS_STOP(&vpack->tw) &&
-      (vpack->tw.durable_stop_ts > rollback_timestamp || prepared)) {
-        /*
-         * Clear the remove operation from the key by inserting the original on-disk value as a
-         * standard update.
-         */
-        WT_RET(__wt_page_cell_data_ref(session, page, vpack, &buf));
-
-        WT_ERR(__wt_upd_alloc(session, &buf, WT_UPDATE_STANDARD, &upd, NULL));
-        upd->txnid = vpack->tw.start_txn;
-        upd->durable_ts = vpack->tw.durable_start_ts;
-        upd->start_ts = vpack->tw.start_ts;
-        F_SET(upd, WT_UPDATE_RESTORED_FROM_DS);
-        WT_STAT_CONN_DATA_INCR(session, txn_rts_keys_restored);
-        __wt_verbose(session, WT_VERB_RECOVERY_RTS(session),
-          "key restored with commit timestamp: %s, durable timestamp: %s txnid: %" PRIu64
-          "and removed commit timestamp: %s, durable timestamp: %s, txnid: %" PRIu64
-          ", prepared: %s",
-          __wt_timestamp_to_string(upd->start_ts, ts_string[0]),
-          __wt_timestamp_to_string(upd->durable_ts, ts_string[1]), upd->txnid,
-          __wt_timestamp_to_string(vpack->tw.stop_ts, ts_string[2]),
-          __wt_timestamp_to_string(vpack->tw.durable_stop_ts, ts_string[3]), vpack->tw.stop_txn,
-          prepared ? "true" : "false");
-    } else
-        /* Stable version according to the timestamp. */
-        return (0);
-
-err:
-    __wt_buf_free(session, &buf);
-    __wt_free(session, upd);
-    return (ret);
-}
-
-/*
- * __rollback_abort_row_ondisk_kv --
- *     Fix the on-disk row K/V version according to the given timestamp.
- */
-static int
-__rollback_abort_row_ondisk_kv(
-  WT_SESSION_IMPL *session, WT_PAGE *page, WT_ROW *rip, wt_timestamp_t rollback_timestamp)
-{
-    WT_CELL_UNPACK_KV *vpack, _vpack;
-    WT_DECL_RET;
-    WT_ITEM buf;
-    WT_UPDATE *upd;
-    char ts_string[5][WT_TS_INT_STRING_SIZE];
-    bool prepared;
+    /*
+     * Assert an exclusive or for rip and cip such that either only a cip for a column store or a
+     * rip for a row store are passed into the function.
+     */
+    WT_ASSERT(session, (rip != NULL && cip == NULL) || (rip == NULL && cip != NULL));
 
-    vpack = &_vpack;
-    WT_CLEAR(buf);
-    upd = NULL;
+    if (rip != NULL) {
```

Comment:
Don't need brackets on this if now that we've removed the asset 👍

Reply:
Fixed.